demos/provider-mock/src/api.ts

```diff
@@ -35,9 +35,12 @@ export function prosopoRouter(): Router {
         try {
             parsed = VerifySolutionBody.parse(req.body)
         } catch (err) {
-            // TODO fix error handling
-            console.log('parsing error')
-            return next(new ProsopoApiError(err as Error, undefined, 400))
+            return next(
+                new ProsopoApiError('CAPTCHA.PARSE_ERROR', {
+                    context: { error: err, errorCode: 400 },
+                    logLevel: 'info',
```

Comment:
Log level determines the log level of the output message. It defaults to error.

If a logger with a log level set to, e.g. error, is passed in, then only errors with logLevel: "error" or above will actually print.

e.g.

const logger = new Logger(logLevel: "error")
new ProsopoError(error_code, {logLevel: 'info', logger}) <-- This will not print anything
new ProsopoError(error_code, {logLevel: 'fatal', logger}) <-- This prints as fatal
new ProsopoError(error_code, {logger}) <-- This prints as error

Reply:
what happens in the case where I've got an error that I don't want printed because I'm going to handle it? e.g. I call a fn which fails due to a failed fetch request. I want to simply retry calling this fn again. The error shouldn't be printed whatsoever

Reply:
then either

1. pass your logger and a log level lower than that logger
2. pass the log level as trace - the default log level is info, so trace won't print anything

Reply:
sure, so proceeding with option 2, if I've got my logger set to trace then it would still print. Do you see my point? there's always a configuration which results in the error being printed. imo we should have the ability not to print anything at all, as some errors should be handled silently

Reply:
sure 2 secs I'll add a silent config option then